pkg/provider/onboardbase/client/client.go

```diff
@@ -0,0 +1,402 @@
+		Project:     request.Project,
+		Environment: request.Environment,
+	}
+
+	secretsData, _, err := c.makeGetSecretsRequest(secretsrequest)
+	if err != nil {
+		return err
+	}
+	secrets, err := c.mapSecretsByPlainKey(secretsData.Data)
+	if err != nil {
+		return err
+	}
+	secret, ok := secrets[request.Name]
+	if !ok || secret.Id == "" {
+		return nil
+	}
+
+	params := request.buildQueryParams()
+	deleteSecretDto := &DeleteSecretsRequest{
+		SecretId: secret.Id,
+	}
+	body, jsonErr := json.Marshal(deleteSecretDto)
+	if jsonErr != nil {
+		return &APIError{Err: jsonErr, Message: "unable to unmarshal delete secrets payload"}
+	}
+	_, err = c.performRequest("/secrets", "DELETE", headers{}, params, body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *OnboardbaseClient) makeGetSecretsRequest(request SecretsRequest) (*secretResponseBody, *apiResponse, error) {
+	headers := headers{}
+
+	params := request.buildQueryParams()
+	response, apiErr := c.performRequest("/secrets", "GET", headers, params, httpRequestBody{})
+	if apiErr != nil {
+		return nil, nil, apiErr
+	}
+
+	var data *secretResponseBody
+	if err := json.Unmarshal(response.Body, &data); err != nil {
+		return nil, nil, &APIError{Err: err, Message: "unable to unmarshal secret payload", Data: string(response.Body)}
+	}
+	return data, response, nil
+}
+
+func (c *OnboardbaseClient) GetSecrets(request SecretsRequest) (*SecretsResponse, error) {
+	data, response, err := c.makeGetSecretsRequest(request)
+	if err != nil {
+		return nil, err
+	}
+
+	secrets, _ := c.getSecretsFromPayload(data.Data)
+	return &SecretsResponse{Secrets: secrets, Body: response.Body}, nil
+}
+
+func (r *SecretsRequest) buildQueryParams() queryParams {
+	params := queryParams{}
+
+	if r.Project != "" {
+		params["project"] = r.Project
+	}
+
+	if r.Environment != "" {
+		params["environment"] = r.Environment
+	}
+
+	return params
+}
+
+func (r *SecretRequest) buildQueryParams() queryParams {
+	params := queryParams{}
+
+	if r.Project != "" {
+		params["project"] = r.Project
+	}
+
+	if r.Environment != "" {
+		params["environment"] = r.Environment
+	}
+
+	return params
+}
+
+func (c *OnboardbaseClient) performRequest(path, method string, headers headers, params queryParams, body httpRequestBody) (*apiResponse, error) {
+	urlStr := c.BaseURL().String() + path
+	reqURL, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, &APIError{Err: err, Message: fmt.Sprintf("invalid API URL: %s", urlStr)}
+	}
+
+	var bodyReader io.Reader
+	if body != nil {
+		bodyReader = bytes.NewReader(body)
+	} else {
+		bodyReader = http.NoBody
+	}
+
+	req, err := http.NewRequest(method, reqURL.String(), bodyReader)
+	if err != nil {
+		return nil, &APIError{Err: err, Message: "unable to form HTTP request"}
+	}
+
+	if method == "POST" && req.Header.Get("content-type") == "" {
+		req.Header.Set("content-type", "application/json")
+	}
+
+	if req.Header.Get("accept") == "" {
+		req.Header.Set("accept", "application/json")
+	}
```

Comment:
Is there any option where you wouldn't want accept and content type be JSON? I think it's safe to just add those headers.